### PR TITLE
Attention sparse embeddings

### PR DESCRIPTION
**Pull request by generall**
Intro new sparse model based on attention weights. Consider it as an extension of bm25 for short documents.

### CODE REVIEW

File: tests/test_attention_embeddings.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        "In the beginning, the universe was created."
+        " This has made a lot of people very angry and been widely regarded as a bad move.",
+        "It's a truth universally acknowledged that a zombie in possession of brains must be in want of more brains.",
+        "War is peace. Freedom is slavery. Ignorance is strength.",
+        "We're not in Infinity; we're in the suburbs.",
+        "I was a thousand times more evil than thou!",
+        "History is merely a list of surprises... It can only prepare us to be surprised yet again.",
+    ]))
+
+    for result in output:
```
Comment (joein) [inline tests/test_attention_embeddings.py:28]: not really useful test :D

Comment (generall) [reply tests/test_attention_embeddings.py:28]: I am open to suggestions

Comment (joein) [reply tests/test_attention_embeddings.py:28]: I would check some types / shapes / values (e.g. that query values are [1, 1, 1, 1]), etc.

By the way, should not we initialize it as `SparseTextEmbedding(model_name="Qdrant/bm42-all-minilm-l6-v2-attentions")` ?
If we want to initialize it as `SparseTextEmbedding`, then we also need to overload methods like `query_embed` in it.

Other than this, the PR looks ok

Comment (generall) [reply tests/test_attention_embeddings.py:28]: > If we want to initialize it as SparseTextEmbedding, then we also need to overload methods like query_embed in it.

yeah, this is right

**Review by joein — APPROVED** [review]

**Pull request merged**